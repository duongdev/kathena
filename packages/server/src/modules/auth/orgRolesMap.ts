import { Permission as P } from './models/Permission'
import { Role } from './models/Role'

export const staff: Role = {
  name: 'staff',
  priority: 3,
  permissions: [
    P.Hr_Access,
    P.Hr_CreateOrgAccount,
    P.Hr_ListOrgAccounts,
    P.Hr_UpdateOrgAccountStatus,
    P.Academic_CreateAcademicSubject,
    P.Academic_ListAcademicSubjects,
    P.Academic_SetAcademicSubjectPublication,
    P.Academic_UpdateAcademicSubject,
    P.Academic_ListCourses,
    P.Academic_UpdateCourse,
    P.Academic_CreateCourse,
    P.Academic_AddLecturersToCourse,
    P.OrgOffice_ListOrgOffices,
    P.OrgOffice_CreateOrgOffice,
    P.OrgOffice_UpdateOrgOffice,
  ],
}

export const student: Role = {
  name: 'student',
  priority: 4,
  permissions: [P.Academic_ListAcademicSubjects],
}

export const lecturer: Role = {
  name: 'lecturer',
  priority: 4,
  permissions: [P.Academic_ListAcademicSubjects],
}

export const admin: Role = {
  name: 'admin',
  priority: 2,
  permissions: [
    ...staff.permissions,
    P.OrgOffice_CreateOrgOffice,
    P.OrgOffice_UpdateOrgOffice,
  ],
}

export const owner: Role = {
  name: 'owner',
  priority: 1,
  permissions: [...admin.permissions],
}

export const orgRoles = [owner, admin, staff, lecturer, student]
