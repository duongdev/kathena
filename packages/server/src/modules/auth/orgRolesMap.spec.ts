import { orgRoles } from './orgRolesMap'

describe('orgRolesMap', () => {
  it(`should match snapshot`, () => {
    expect(orgRoles).toMatchSnapshot()
  })
})
