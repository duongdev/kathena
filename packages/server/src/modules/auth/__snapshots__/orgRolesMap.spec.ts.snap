// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`orgRolesMap should match snapshot 1`] = `
Array [
  Object {
    "name": "owner",
    "permissions": Array [
      "Hr_Access",
      "Hr_CreateOrgAccount",
      "Hr_ListOrgAccounts",
      "Hr_UpdateOrgAccountStatus",
      "Academic_CreateAcademicSubject",
      "Academic_ListAcademicSubjects",
      "Academic_SetAcademicSubjectPublication",
      "OrgOffice_ListOrgOffices",
      "OrgOffice_CreateOrgOffice",
    ],
    "priority": 1,
  },
  Object {
    "name": "admin",
    "permissions": Array [
      "Hr_Access",
      "Hr_CreateOrgAccount",
      "Hr_ListOrgAccounts",
      "Hr_UpdateOrgAccountStatus",
      "Academic_CreateAcademicSubject",
      "Academic_ListAcademicSubjects",
      "Academic_SetAcademicSubjectPublication",
      "OrgOffice_ListOrgOffices",
      "OrgOffice_CreateOrgOffice",
    ],
    "priority": 2,
  },
  Object {
    "name": "staff",
    "permissions": Array [
      "Hr_Access",
      "Hr_CreateOrgAccount",
      "Hr_ListOrgAccounts",
      "Hr_UpdateOrgAccountStatus",
      "Academic_CreateAcademicSubject",
      "Academic_ListAcademicSubjects",
      "Academic_SetAcademicSubjectPublication",
      "OrgOffice_ListOrgOffices",
    ],
    "priority": 3,
  },
  Object {
    "name": "lecturer",
    "permissions": Array [],
    "priority": 4,
  },
  Object {
    "name": "student",
    "permissions": Array [],
    "priority": 4,
  },
]
`;
